code/modules/hazmat/crystal_maker.dm

```diff
@@ -0,0 +1,20 @@
+/obj/machinery/hazmat
+	icon = 'icons/obj/hazmat/machinery.dmi'
+
+/obj/machinery/hazmat/crystal_maker
+	name = "crystal maker"
+	desc = "Insert diamonds or glass to generate crystals. Uses 1 sheet each."
+	icon_state = "crystal_maker"
+
+/obj/machinery/hazmat/crystal_maker/attackby(obj/item/I, mob/user, params)
+	. = ..()
+	if(istype(I, /obj/item/stack/sheet/glass))
+		var/obj/item/crystal/random/C = new(get_turf(src))
+		C.malf_chance = 10
```

Comment:
magic number

Reply:
its a percent chance, it's not magic

Reply:
yeah but it's coming out of nowhere and you have to dive into the proc to change how much glass percentually affects crystals.

Reply:
ye fair